main/test/ca/uwaterloo/flix/library/TestFloat32.flix

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2017 Liam Palmer
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/////////////////////////////////////////////////////////////////////////////
+// size                                                                    //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def size01: Bool = Float32.size() `assertEq!` 32
+
+/////////////////////////////////////////////////////////////////////////////
+// minExponent                                                             //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def minExponent01: Bool = Float32.minExponent() `assertEq!` -126
+
+/////////////////////////////////////////////////////////////////////////////
+// maxExponent                                                             //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def maxExponent01: Bool = Float32.maxExponent() `assertEq!` 127
+
+/////////////////////////////////////////////////////////////////////////////
+// maxValue                                                                //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def maxValue01: Bool = Float32.maxValue() `assertEq!` ((2.0f32 - 2.0f32 ** -23.0f32) * (2.0f32 ** 127.0f32))
+
+/////////////////////////////////////////////////////////////////////////////
+// minValue                                                                //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def minValue01: Bool = Float32.minValue() `assertEq!` (-(2.0f32 - 2.0f32 ** -23.0f32) * (2.0f32 ** 127.0f32))
+
+/////////////////////////////////////////////////////////////////////////////
+// minPositiveValue                                                        //
+/////////////////////////////////////////////////////////////////////////////
+@test
+def minPositiveValue01: Bool = Float32.minPositiveValue() `assertEq!` (2.0f32 ** -149.0f32)
+
+/////////////////////////////////////////////////////////////////////////////
+// undefined                                                               //
+/////////////////////////////////////////////////////////////////////////////
+//@test
+//def undefined01: Bool = Float32.undefined() `assertEq!` (0.0f32 / 0.0f32)
```

Comment:
This test fails for both Float32 and Float64, even (0.0f32 / 0.0f32) == (0.0f32 / 0.0f32) returns false, I'm not sure why.